src/data/nat/interval.lean

```diff
@@ -174,6 +174,76 @@ begin
     lt_iff_le_and_ne],
 end
 
+lemma mod_inj_on_Ico (n a : ℕ) : set.inj_on (λ k, k % a) (finset.Ico n (n+a)) :=
+begin
+  induction n with n ih,
+  { simp only [zero_add, nat_zero_eq_zero, Ico_zero_eq_range],
+    rintro k hk l hl (hkl : k % a = l % a),
+    simp only [finset.mem_range, finset.mem_coe] at hk hl,
+    rwa [mod_eq_of_lt hk, mod_eq_of_lt hl] at hkl, },
+  rw [Ico_succ_left_eq_erase_Ico, succ_add, Ico_succ_right_eq_insert_Ico le_self_add],
+  rintro k hk l hl (hkl : k % a = l % a),
+  have ha : 0 < a,
+  { by_contra ha, simp only [not_lt, nonpos_iff_eq_zero] at ha, simpa [ha] using hk },
+  simp only [finset.mem_coe, finset.mem_insert, finset.mem_erase] at hk hl,
+  rcases hk with ⟨hkn, (rfl|hk)⟩; rcases hl with ⟨hln, (rfl|hl)⟩,
+  { refl },
+  { rw add_mod_right at hkl,
+    suffices : l = n, { contradiction },
+    refine ih hl _ hkl.symm,
+    simp only [lt_add_iff_pos_right, set.left_mem_Ico, finset.coe_Ico, ha], },
+  { rw add_mod_right at hkl,
+    suffices : k = n, { contradiction },
+    refine ih hk _ hkl,
+    simp only [lt_add_iff_pos_right, set.left_mem_Ico, finset.coe_Ico, ha], },
+  { refine ih _ _ hkl; simp only [finset.mem_coe, hk, hl], },
+end
+
+
+-- what is the correct generality? It also works for `ℤ`
+lemma finset_Ico_image_mod (n a : ℕ) :
+  (Ico n (n+a)).image (λ k, k % a) = range a :=
+begin
+  obtain (rfl|ha) : a = 0 ∨ a ≠ 0 := eq_or_ne a 0,
+  { rw [range_zero, add_zero, Ico_self, image_empty], },
+  ext i,
+  simp only [mem_image, exists_prop, mem_range, mem_Ico],
+  split,
+  { rintro ⟨i, h, rfl⟩, exact mod_lt i ha.bot_lt },
+  { intro hia,
+    have hn := nat.mod_add_div n a,
+    by_cases hi : i < n % a,
+    { refine ⟨i + a * (n/a + 1), ⟨_, _⟩, _⟩,
+      { rw [add_comm (n/a), mul_add, mul_one, ← add_assoc],
+        refine hn.symm.le.trans (add_le_add_right _ _),
+        simpa only [zero_add] using add_le_add (zero_le i) (nat.mod_lt n ha.bot_lt).le, },
+      { refine lt_of_lt_of_le (add_lt_add_right hi (a * (n/a + 1))) _,
+        rw [mul_add, mul_one, ← add_assoc, hn], },
+      { rw [nat.add_mul_mod_self_left, nat.mod_eq_of_lt hia], } },
+    { refine ⟨i + a * (n/a), ⟨_, _⟩, _⟩,
+      { push_neg at hi,
+        exact hn.symm.le.trans  (add_le_add_right hi _), },
+      { rw [add_comm n a],
+        refine add_lt_add_of_lt_of_le hia (le_trans _ hn.le),
+        simp only [zero_le, le_add_iff_nonneg_left], },
+      { rw [nat.add_mul_mod_self_left, nat.mod_eq_of_lt hia], } } }
```

Comment:
You can unindent all that.
```suggestion
  intro hia,
  have hn := nat.mod_add_div n a,
  by_cases hi : i < n % a,
  { refine ⟨i + a * (n/a + 1), ⟨_, _⟩, _⟩,
    { rw [add_comm (n/a), mul_add, mul_one, ← add_assoc],
      refine hn.symm.le.trans (add_le_add_right _ _),
      simpa only [zero_add] using add_le_add (zero_le i) (nat.mod_lt n ha.bot_lt).le, },
    { refine lt_of_lt_of_le (add_lt_add_right hi (a * (n/a + 1))) _,
      rw [mul_add, mul_one, ← add_assoc, hn], },
    { rw [nat.add_mul_mod_self_left, nat.mod_eq_of_lt hia], } },
  { refine ⟨i + a * (n/a), ⟨_, _⟩, _⟩,
    { push_neg at hi,
      exact hn.symm.le.trans  (add_le_add_right hi _), },
    { rw [add_comm n a],
      refine add_lt_add_of_lt_of_le hia (le_trans _ hn.le),
      simp only [zero_le, le_add_iff_nonneg_left], },
    { rw [nat.add_mul_mod_self_left, nat.mod_eq_of_lt hia], } }
```

Reply:
Doesn't the style guide tell us that goals that arise from previous tactics should be brace-enclosed and indented?